### ISSUE TITLE
Error decoding flac files in multi-threaded mode

### ISSUE THREAD

**Issue by tyler-nervana** [NervanaSystems/aeon#18]
When using a manifest file containing .flac files, the audio transformer segfaults with this error: 

```
formats: no handler for detected file type `flac'

Program received signal SIGSEGV, Segmentation fault.
```

This only happens if the dataloader uses multiple decoding threads. Initializing with `single_thread=True` seems to work reliably. The error can be traced to this line in `loader/src/utils.cpp`: `sox_format_t* in = sox_open_mem_read((void *) item, itemSize, NULL, NULL, NULL);`

Below is a simple python script that should reproduce the issue:

``` python
import tempfile
import numpy as np
from neon.backends import gen_backend
from aeon.dataloader import DataLoader

batch_size = 10
filenames = ["/path/to/audio.flac"]

manifest_file = tempfile.mktemp(prefix="manifest")
lines = "\n".join(filenames * int(np.ceil(batch_size / len(filenames))))
with open(manifest_file, "w") as manifest:
    manifest.write(lines)

aeon_backend = gen_backend("cpu")
aeon_backend.bsz = batch_size

max_utt_len = 15000
sampling_freq = 16000
feature_type = "samples"

aeon_audio = dict(sample_freq_hz=sampling_freq,
                  max_duration="{} seconds".format(max_utt_len / 1000.),
                  frame_length="1 samples",
                  frame_stride="1 samples",
                  feature_type=feature_type,
                  output_type="float")

aeon_config = dict(type="audio",
                   audio=aeon_audio,
                   manifest_filename=manifest_file,
                   minibatch_size=batch_size,
                   macrobatch_size=batch_size,
                   shuffle_manifest=False,
                   single_thread=False)

print("Initializing the dataloader in multi-threaded mode")
aeon_config["single_thread"] = False
aeon_dl = DataLoader(backend=aeon_backend,
                     config=aeon_config)
aeon_item = list(aeon_dl.next())
```

**Comment by tyler-nervana** [comment]
Interestingly, if you first initialize a dataloader with `single_thread=True` and then start one with `single_thread=False` in the same script, it seems to run reliably. Hopefully this will suggest a fix or at least a workaround.

**Comment by apark263** [comment]
should be fixed with 46ca5374
